Doc/library/typing.rst

```diff
@@ -1754,15 +1754,27 @@ without the dedicated syntax, as documented below.
          the constraints are evaluated only when the attribute is accessed, not when
          the type variable is created (see :ref:`lazy-evaluation`).
 
+   .. attribute:: __default__
+
+      The default value of the type variable, if any. This attribute is ``None``
```

Comment:
I think an example should be definitely helpful because people might really be confused between `TypeVar('T')` and `TypeVar('T', default=None)`.

Reply:
(FWIW, I don't think that design question that I raised on Discourse should block the initial implementation here being merged.)

Reply:
I don't think it should be a blocker, but I support your suggestion on discourse. 

Now, whatever is decided, the representation shouldn't change too many times afterwards. I think a similar issue (but I don't know whether it was formalized or not) is about the `<not given>` for enumerations, where we had some different representations across versions. This could definitely be some use cases for [PEP 661](https://peps.python.org/pep-0661/) (but I don't know if this one is actively needed). And depending on how you backports things, having a public value known to be the sentinel is better when you test representation outputs (the sentinel for enumerations is private).

Reply:
For BC reasons we probably can't afford to change this behavior later. I guess we can merge this version first and tweak the implementation during the beta phase, though.

Reply:
> For BC reasons we probably can't afford to change this behavior later. I guess we can merge this version first and tweak the implementation during the beta phase, though.

yes, that was what I was thinking

Reply:
Maybe it's obvious but what does "BC" mean in this context?

Reply:
Backwards Compatibility